File: app/src/main/java/com/erebor/tomkins/pos/helper/CheckPermissionHelper.java
package com.erebor.tomkins.pos.helper;

import android.Manifest;
import android.app.Activity;
import android.content.pm.PackageManager;

import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;

/**
 * Created by ryandzhunter on 07/03/18.
 */

public class CheckPermissionHelper {

    private Activity activity;

    public CheckPermissionHelper(Activity activity) {
        this.activity = activity;
    }

    public boolean checkCameraPermission(){
        return ContextCompat.checkSelfPermission(activity, Manifest.permission.CAMERA) == PackageManager.PERMISSION_GRANTED;
    }

    public boolean checkCameraAndExternalStoragePermission(){
        return ContextCompat.checkSelfPermission(activity, Manifest.permission.WRITE_EXTERNAL_STORAGE)
                == PackageManager.PERMISSION_GRANTED && ContextCompat.checkSelfPermission(activity,
                Manifest.permission.CAMERA) == PackageManager.PERMISSION_GRANTED;
    }

    public void showCameraPermission(int code){
        ActivityCompat.requestPermissions(activity, new String[]{Manifest.permission.CAMERA},
                code);
    }

    public void showCameraAndExternalPermission(int code){
        ActivityCompat.requestPermissions(activity,
                new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.CAMERA},
                code);
    }

}

File: app/src/main/java/com/erebor/tomkins/pos/helper/CheckPermissionHelper.kt
package com.erebor.tomkins.pos.helper

import android.Manifest
import android.app.Activity
import android.content.pm.PackageManager

import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat

/**
 * Created by ryandzhunter on 07/03/18.
 */

class CheckPermissionHelper(private val activity: Activity) {

    fun checkCameraPermission(): Boolean {
        return ContextCompat.checkSelfPermission(activity, Manifest.permission.CAMERA) == PackageManager.PERMISSION_GRANTED
    }

    fun checkCameraAndExternalStoragePermission(): Boolean {
        return ContextCompat.checkSelfPermission(activity, Manifest.permission.WRITE_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED && ContextCompat.checkSelfPermission(activity,
                Manifest.permission.CAMERA) == PackageManager.PERMISSION_GRANTED
    }

    fun showCameraPermission(code: Int) {
        ActivityCompat.requestPermissions(activity, arrayOf(Manifest.permission.CAMERA),
                code)
    }

    fun showCameraAndExternalPermission(code: Int) {
        ActivityCompat.requestPermissions(activity,
                arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.CAMERA),
                code)
    }

}
